### PR TITLE
PYTHON-4883 Add release date to changelog entries

### PR DESCRIPTION
**Pull request by aclark4life**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -794,8 +794,14 @@ in this release.
 .. _PYTHON-3186: https://jira.mongodb.org/browse/PYTHON-3186
 .. _pymongo-stubs: https://github.com/mongodb-labs/pymongo-stubs
 
-Changes in Version 4.0
-----------------------
+Changes in Version 4.0.2 (2022/03/03)
+-------------------------------------
```
Comment (blink1073) [inline doc/changelog.rst:798]: There were no commits in https://github.com/mongodb/mongo-python-driver/compare/v4.0...4.0.2, so I think we should just say no commits due to lack of historical context.

**Review by blink1073 — CHANGES_REQUESTED** [review]

**Review by blink1073 — APPROVED** [review]
LGTM!

**Comment by blink1073** [comment]
Ah, lint:

```
doc/changelog.rst:390: D000 Title underline too short.
doc/changelog.rst:390: D000 Title underline too short.
```

**Review by blink1073 — APPROVED** [review]

**Comment by aclark4life** [comment]
Thanks!

**Pull request merged**